fastn-core/src/http.rs

```diff
@@ -788,6 +794,45 @@ pub fn user_err(
         .body(serde_json::to_string(&resp)?))
 }
 
+/// Checks whether a request was made by a Google/Bing bot based on its User-Agent
+/// Google crawlers and fetchers: https://developers.google.com/search/docs/crawling-indexing/overview-google-crawlers
+/// Bing crawlers: https://www.bing.com/webmasters/help/which-crawlers-does-bing-use-8c184ec0
+pub fn is_bot(user_agent: &str) -> bool {
```

Comment:
Let's also include popular social media sites. We have to ensure they have access to social media graph meta data at least, not the full HTML, so if we can do that without doing full SSR that is also fine. @Arpita-Jaiswal is that possible?